graphql/resolvers/roomResolver.js

```diff
@@ -3,64 +3,101 @@ const Service = require("../../models/service/service");
 const AccessGroupRoom = require("../../models/accessgrouproom/accessgrouproom");
 const Access = require("../../models/accessgroup/accessgroup");
 const _ = require("lodash");
+const AccessGroupUser = require("../../models/accessGroupUser/accessgroupuser");
 
+const getAccessGroupRoom = async user => {
+  accessGroupUser = await AccessGroupUser.find({ userId: user.id });
+  if (!accessGroupUser) {
+    throw new Error("talk with admin, no access");
+  }
+
+  let accessGroupRoom = await AccessGroupRoom.find({
+    accessGroupId: { $in: accessGroupUser.map(item => item.accessGroupId) }
+  });
+  if (!accessGroupRoom) {
+    throw new Error(
+      "talk with admin, issue with your access & room connection"
+    );
+  }
+  return accessGroupRoom;
+};
 
 const resolver = {
   Query: {
-    rooms: () => Room.find({}),
-    room: async (_parent, args) => Room.findById({ _id: args.id }),
-    roomByService: (_parent, args) => Room.find({ serviceId: args.id }),
-    roomsWithAccessGroup: async (_parent, args) => {
-      let list = await AccessGroupRoom.find({ accessGroupId: args.id });
-      return Room.find({
-        _id: { $in: list.map(item => item.roomId) }
+    rooms: async (parent, booking, { user }) => {
+      if (!user) {
+        throw new Error("not authorized");
+      }
+
+      if (user.admin) {
+        return Room.find({});
+      }
+      let accessGroupRoom = getAccessGroupRoom(user);
+
+      return await Room.find({
+        _id: { $in: accessGroupRoom.map(item => item.roomId) }
```

Comment:
error: 
"accessGroupRoom.map is not a function"
Occurs when user is not an admin.
Error independent of if the user has accessGroups, or not[ ]

fix:
let accessGroupRoom = await getAccessGroupRoom(user);

Reply:
fixed